packages/ui/src/components/va-split/VaSplit.vue

```diff
@@ -0,0 +1,281 @@
+          return 100
+        default:
+          warn('Invalid limits measure!')
+          return 0
+      }
+    }
+    const getPanelMinMax = (v: SplitLimit) => {
+      if (!v || !containerSize.value) { return }
+
+      let minPercents = 0
+      let maxPercents = 100
+
+      if (isString(v) || isNumber(v)) { minPercents = convertToPercents(v) }
+
+      if (Array.isArray(v)) {
+        minPercents = convertToPercents(v[0])
+        maxPercents = convertToPercents(v[1])
+      }
+
+      if (minPercents > maxPercents) {
+        warn(`Min panels size can not be larger than max one! Passed limit: ${v}.`)
+        maxPercents = minPercents
+      }
+
+      return { min: minPercents, max: maxPercents }
+    }
+
+    const startPanelMinMax = computed(() => getPanelMinMax(props.limits[0]))
+    const endPanelMinMax = computed(() => {
+      const result = getPanelMinMax(props.limits[1])
+
+      // here potentially can be a conflict between two checks below but the first one won't break component
+      if (result?.min && startPanelMinMax.value?.min && Math.floor(result.min + startPanelMinMax.value.min) > 100) {
+        warn('The sum of different panels min sizes should be lesser than 100% of the container size!')
+        result.min = 100 - startPanelMinMax.value.min
+      }
+
+      if (result?.min && result?.max &&
+        startPanelMinMax.value?.min && startPanelMinMax.value?.max &&
+        (Math.ceil(result.min + startPanelMinMax.value.max) < 100 ||
+          Math.ceil(result.max + startPanelMinMax.value.min) < 100)) {
+        warn('The sum of different panels min and max sizes should be equal to 100% of the container size!')
+        result.min = 100 - startPanelMinMax.value.max
+      }
+
+      return result
+    })
+
+    const splitterPosition = ref(props.modelValue)
+    const splitterPositionComputed = computed(() => {
+      if (!startPanelMinMax.value || !endPanelMinMax.value) { return splitterPosition.value }
+
+      // checking min panels size
+      if (splitterPosition.value < startPanelMinMax.value.min) { return startPanelMinMax.value.min }
+      if (splitterPosition.value > 100 - endPanelMinMax.value.min) { return 100 - endPanelMinMax.value.min }
+
+      // checking max panels size
+      if (splitterPosition.value > startPanelMinMax.value.max) { return startPanelMinMax.value.max }
+      if (splitterPosition.value < 100 - endPanelMinMax.value.max) { return 100 - endPanelMinMax.value.max }
+
+      return splitterPosition.value
+    })
+
+    const {
+      isDragging,
+      startDragging,
+      processDragging,
+      stopDragging,
+      currentSplitterPosition,
+    } = useSplitDragger(containerSize, splitterPositionComputed, props)
+
+    const maximizePanel = () => {
+      if (!props.maximization || props.disabled || !startPanelMinMax.value || !endPanelMinMax.value) { return }
+
+      splitterPosition.value = props.maximizeStart ? 100 - endPanelMinMax.value.min : startPanelMinMax.value.min
```

Comment:
```suggestion
      splitterPosition.value = props.maximizeStart ? startPanelMinMax.value.max : 100 - endPanelMinMax.value.max
```

Reply:
Check again. It's correct logic now.

Reply:
Absolutely the same result. But ok, `max` looks more appropriate.